server/src/main/java/org/elasticsearch/action/datastreams/GetDataStreamAction.java

```diff
@@ -331,6 +364,9 @@ public XContentBuilder toXContent(
                 builder.field(ALLOW_CUSTOM_ROUTING.getPreferredName(), dataStream.isAllowCustomRouting());
                 builder.field(REPLICATED.getPreferredName(), dataStream.isReplicated());
                 builder.field(ROLLOVER_ON_WRITE.getPreferredName(), dataStream.rolloverOnWrite());
+                if (this.maximumTimestamp != null) {
+                    builder.field(MAXIMUM_TIMESTAMP.getPreferredName(), this.maximumTimestamp);
```

Comment:
A human readable form would be really nice here if this is potentially going to be consumed by anything other than a UI. It doesn't look like we support that in stats, but they don't necessarily have to be the same, right? (And maybe we ought to add that to stats, too)

Reply:
Actually forget that last comment -- I don't think we have a human readable transformer for epoch millis.